lib/dialogs/buy/buy_dialog.dart

```diff
@@ -279,11 +279,22 @@ class _BuyDialogState extends State<BuyDialog> {
     );
 
     return BlocConsumer<BuyDialogBloc, BuyDialogState>(
-      listenWhen: (_, current) => current.status == BlocStatus.error,
-      listener: (context, state) => context.showWarningToast(
-        title: 'Action Error',
-        description: 'An error has occured while handling the Buy action',
-      ),
+      listenWhen: (_, current) =>
+          current.status == BlocStatus.error ||
+          current.status == BlocStatus.noData,
+      listener: (context, state) {
+        if (state.status == BlocStatus.noData) {
+          context.showWarningToast(
+            title: 'No Pool Data',
```

Comment:
Add better messaging here. The current message could potentially confuse the user into thinking that this is the pool page.